### PR TITLE
Physics study fixes

### PR DESCRIPTION
**Pull request by kimberscott**
Display a more accurate session count to participants at the end of the physics study. Fix counterbalancing using one of the first 6 object mappings in the lists so it now chooses a random starting point.

### CODE REVIEW

**Comment by pattisdr** [comment]
Reviewing now...

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Your computed property is actually relying on pastSessions, so I would do 
``` 
currentSessionStatus: Ember.computed('frameContext.pastSessions', function() {
```

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Can this be removed?

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: It's inefficient to loop through all these sessions twice, both for expData and expDates.

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Ember uses ES2015 syntax which has a lot of improvements over plain vanilla javascript. You should use [forEach](https://guides.emberjs.com/v2.16.0/object-model/enumerables/#toc_iterating-over-an-enumerable) to enumerate over the variables in an array

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Use [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) instead of this old method `.indexOf`

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: forEach

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: [const instead of var](http://putaindecode.io/en/articles/js/es2015/const-let-var/). If the variable is not going to change, use const.

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: [`let` instead of `var`](http://putaindecode.io/en/articles/js/es2015/const-let-var/).  Variable is going to change so use let. there are other instances of this too.

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: In general, I wouldn't define variables you're only using once.

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Not a fan of returning arrays, and later accessing the first or second item in the array. The context of what you are doing gets lost.    I would return a dictionary here:
```
        return {'nSessions': nSessions, 'daysSessionsCompleted': moment.max(sessionDates).diff(moment.min(sessionDates), 'days') + 1 };
```

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Instead of accessing first item in array, access appropriate dictionary item:
```
return this.get('currentSessionStatus.nSessions');
```

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js]: Instead of accessing first item in array, access appropriate dictionary item:
```
return this.get('currentSessionStatus.daysSessionsCompleted');
```

File: exp-player/addon/randomizers/pref-phys.js
Comment (pattisdr) [inline exp-player/addon/randomizers/pref-phys.js]: Can you explain to me what you are doing here?

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [reply exp-player/addon/components/exp-exit-survey/component.js]: I think you can use const for arrays even if you are adding items to arrays.  Might not hold up if you were changing an array to a string.

File: exp-player/addon/components/exp-exit-survey/component.js
Comment (pattisdr) [inline exp-player/addon/components/exp-exit-survey/component.js:127]: More explanation below, but here's how I would write the currentSessionStatus. It's more condensed and you don't have to loop through all your sessions twice in the beginning.  You only have to fetch the session's createdOn if that session had the min number of physics videos taken.  Not defining variables if they're only going to be used once. Uses some more modern ES6 JS formatting. Returns a dictionary instead of an array.

``` js
   currentSessionStatus: Ember.computed('frameContext.pastSessions', function() {
        let nSessions = 0;
        const sessionDates = [];
        this.get('frameContext.pastSessions').forEach(session => {
            let nVideos = 0;
            Object.keys(session.get('expData')).forEach(frameKeyName => {
                if (frameKeyName.includes('pref-phys-videos')) {
                    nVideos++;
                }
            });
            // Count only sessions with at least minVideos pref-phys-videos frames
            if (nVideos >= this.get('minVideosToCountSession')) {
                nSessions++;
                sessionDates.pushObject(moment(session.get('createdOn')));
            }
        });
        return {'nSessions': nSessions, 'daysSessionsCompleted': moment.max(sessionDates).diff(moment.min(sessionDates), 'days') + 1 };
    }),

```

**Review by pattisdr — CHANGES_REQUESTED** [review]
Hey @kimberscott, review finished, added some comments for your general knowledge, but pasted a blurb of how I would write the currentSessionStatus and then you would need to make changes to currentSessionsCompleted and currentDaysSessionsCompleted to accommodate.

Nice addition on trying to get users to get through at least 6 videos.

**Comment by kimberscott** [comment]
Thank you so much again for all the help on this, and the explanations of better practices that I can apply elsewhere as well. I think this one should also be set to merge.

**Comment by pattisdr** [comment]
Looks good to me, thanks for the explanations on the randomizer portion. Will merge when tests are passing.

**Review by pattisdr — APPROVED** [review]

**Pull request merged**
